### PR TITLE
docs(examples): Added Auth.js 5 example app

### PR DESCRIPTION
**Pull request by davidmytton**
This example shows how to use an Arcjet rate limit with a user ID from [Auth.js authentication with Next.js](https://authjs.dev/). It's a copy of [the Next.js demo](https://github.com/nextauthjs/next-auth/tree/5ea8b7b0f4d285e48f141dd91e518c905c9fb34e/apps/examples/nextjs), but with Arcjet added.

### CODE REVIEW

File: examples/nextjs-14-authjs-5/package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "private": true,
+  "description": "An example project for NextAuth.js with Next.js",
+  "repository": "https://github.com/nextauthjs/next-auth-example.git",
+  "bugs": {
+    "url": "https://github.com/nextauthjs/next-auth/issues"
+  },
+  "homepage": "https://next-auth-example.vercel.app",
```
Comment (blaine-arcjet) [inline examples/nextjs-14-authjs-5/package.json:8]:
```suggestion
```
Let's chop out some of this stuff that doesn't point to us

File: examples/nextjs-14-authjs-5/package.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    "build": "next build",
+    "start": "next start"
+  },
+  "author": "Iain Collins <me@iaincollins.com>",
+  "contributors": [
+    "Balázs Orbán <info@balazsorban.com>",
+    "Nico Domino <yo@ndo.dev>",
+    "Lluis Agusti <hi@llu.lu>",
+    "Thang Huu Vu <hi@thvu.dev>"
+  ],
```
Comment (blaine-arcjet) [inline examples/nextjs-14-authjs-5/package.json:20]:
```suggestion
```

File: examples/nextjs-14-authjs-5/package.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    "@radix-ui/react-slot": "^1.0.2",
+    "class-variance-authority": "^0.7.0",
+    "clsx": "^2.0.0",
+    "lucide-react": "^0.274.0",
+    "next": "latest",
+    "next-auth": "beta",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "tailwind-merge": "^1.14.0",
+    "tailwindcss-animate": "^1.0.7"
```
Comment (blaine-arcjet) [inline examples/nextjs-14-authjs-5/package.json:35]: I'm not happy that we're bringing on a bunch more dependencies here but I'll just close my eyes.

**Review by blaine-arcjet — APPROVED** [review]
Generally looks good. I'm excited for the day where we can gut a bunch of the dependencies these sample projects take on.

File: examples/nextjs-14-authjs-5/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+  "private": true,
+  "description": "An example project for NextAuth.js with Next.js",
+  "scripts": {
+    "dev": "next",
+    "build": "next build",
+    "start": "next start"
+  },
+  "dependencies": {
+    "@radix-ui/react-avatar": "^1.0.3",
```
Comment (blaine-arcjet) [inline examples/nextjs-14-authjs-5/package.json:10]:
```suggestion
    "@arcjet/next": "file:../../arcjet-next",
    "@radix-ui/react-avatar": "^1.0.3",
```
I think this is missing

**Comment by blaine-arcjet** [comment]
I realized this and the other example weren't added to CI

**Review by blaine-arcjet — CHANGES_REQUESTED** [review]
It looks like this is failing to build in CI. @davidmytton can you take a look?

**Comment by davidmytton** [comment]
@blaine-arcjet builds are now working.

File: .gitignore [hunk 5]
```diff
@@ -73,7 +73,6 @@ web_modules/
 .yarn-integrity
 
 # dotenv environment variable files
-.env
```
Comment (blaine-arcjet) [inline .gitignore:76]: I think this was a holdover removal from experimentation. I don't see a `.env` file so we should add it back to this file.

File: examples/nextjs-13-pages-wrap/package-lock.json
Comment (blaine-arcjet) [inline examples/nextjs-13-pages-wrap/package-lock.json:1]: This is unrelated to this PR so we should remove it. Dependabot handles the updates for this example and I don't trust `npm update`

File: .gitignore (hunk 5 above)
Comment (davidmytton) [reply .gitignore:76]: We could, but I kept it because [the NextJS docs recommend](https://nextjs.org/docs/app/building-your-application/configuring/environment-variables) allowing them to be committed.

> Good to know: .env, .env.development, and .env.production files should be included in your repository as they define defaults. .env*.local should be added to .gitignore, as those files are intended to be ignored. .env.local is where secrets can be stored.

File: examples/nextjs-13-pages-wrap/package-lock.json
Comment (davidmytton) [reply examples/nextjs-13-pages-wrap/package-lock.json:1]: The builds were failing on this PR for that example, so I updated it to make them pass following others on https://github.com/vercel/next.js/discussions/30468 and in the recommendation at https://nextjs.org/docs/messages/failed-loading-swc#possible-ways-to-fix-it

We've already implemented the minify opt-out and it hasn't fixed it. 🫠

Comment (blaine-arcjet) [reply examples/nextjs-13-pages-wrap/package-lock.json:1]: This update won't fix it, as it is just a test flake. In Next.js 13, they don't install swc via npm and instead try to download and unzip it which fails all the time. As seen in your commits after the `npm update`, the next-13 test continues to fail. We just need to accept that it's a flakey test or remove the next13 example.

File: .gitignore (hunk 5 above)
Comment (blaine-arcjet) [reply .gitignore:76]: Oh man, that advice is going to get a lot of folks in trouble 🙈 I've always followed the guidance not to check in a .env file.

Comment (davidmytton) [reply .gitignore]: Yeh that's better advice. I've reverted this.

File: examples/nextjs-13-pages-wrap/package-lock.json
Comment (davidmytton) [reply examples/nextjs-13-pages-wrap/package-lock.json:1]: Fair enough. I've reverted this.

**Review by blaine-arcjet — APPROVED** [review]
